### PR TITLE
🐛 amp-next-page: Fix rendering within shadow documents

### LINKED ISSUE #16666: amp-next-page not working with amp-shadow-doc-loader
## What's the issue?

amp-next-page is not working when using the shadow dom

## How do we reproduce the issue?
When loading a amp shadow that contains amp-next-pages the next-page service of the current window is an empty service as we should check the services of the current shadow dom instead of the window.

screen shot 2018-07-10 at 17 41 55
## What browsers are affected?

All browsers

## Which AMP version is affected?

Version 1530567757224

### PR DESCRIPTION
**Pull request by peterjosling**
If `amp-next-page` tries to load in a shadow document, the service gets registered against the AmpDoc rather than the window, and `getService()` tries to read it from the window.

Fixes #16666

### CODE REVIEW

**Comment by peterjosling** [comment]
@aghassemi

File: extensions/amp-next-page/0.1/amp-next-page.js [hunk 1]
```diff
@@ -72,38 +64,45 @@ export class AmpNextPage extends AMP.BaseElement {
...
-      const scriptElement = scriptElements[0];
-      user().assert(isJsonScriptTag(scriptElement),
-          `${TAG} config should ` +
-          'be inside a <script> tag with type="application/json"');
-      const configJson = tryParseJson(scriptElement.textContent, error => {
-        user().error(TAG, 'failed to parse config', error);
-      });
-      this.register_(configJson, separator);
-    }
+    getServicePromiseForDoc(this.getAmpDoc(), SERVICE_ID).then(service => {
```
Comment (aghassemi) [inline extensions/amp-next-page/0.1/amp-next-page.js]: Please `return` the promise to eliminate any possibility of a race between `buildCallback` and `layoutCallback` (it is a little know fact that `buildCallback` can return a `promise`)

Comment (peterjosling) [reply extensions/amp-next-page/0.1/amp-next-page.js]: Oh cool! Done.

**Comment by aghassemi** [comment]
@peterjosling Just some unit test failures related to the component. Will merge when green. Thanks!

**Comment by peterjosling** [comment]
Done.

**Pull request merged**